client/src/context/useProfileContext.tsx

```diff
@@ -0,0 +1,48 @@
+import { useState, useContext, createContext, FunctionComponent, useEffect, useCallback } from 'react';
+// import { useHistory } from 'react-router-dom';
+import { ProfileApiData, ProfileApiDataSuccess } from '../interface/ProfileApiData';
+import { UserProfile } from '../interface/Profile';
+import fetchUserProfile from '../helpers/APICalls/fetchUserProfile';
+
+interface IProfileContext {
+  currentUserProfile: UserProfile | null | undefined;
+  updateProfileContext: (data: ProfileApiDataSuccess) => void;
+}
+
+export const ProfileContext = createContext<IProfileContext>({
+  currentUserProfile: undefined,
+  updateProfileContext: () => null,
+});
+
+export const ProfileProvider: FunctionComponent = ({ children }): JSX.Element => {
+  const [currentUserProfile, setCurrentUserProfile] = useState<UserProfile | null | undefined>();
+
+  // update current user's profile
+  const updateProfileContext = useCallback((data: ProfileApiDataSuccess) => {
+    setCurrentUserProfile(data.profile);
+  }, []);
+
+  useEffect(() => {
+    console.log('user profile context running in contect page');
```

Comment:
Please be mindful of unnecessary console logs in the project, this isn't necessary.